homeassistant/components/light/opple.py

```diff
@@ -0,0 +1,130 @@
+"""
+Support for the Opple light.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.opple/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR_TEMP, Light)
+from homeassistant.const import CONF_HOST, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.color import \
+    color_temperature_kelvin_to_mired as kelvin_to_mired
+from homeassistant.util.color import \
+    color_temperature_mired_to_kelvin as mired_to_kelvin
+
+REQUIREMENTS = ['pyoppleio==1.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default="opple light"): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Opple light platform."""
+    name = config.get('name')
+    host = config.get('host')
+    add_devices([OppleLight(name, host)])
+
+
+class OppleLight(Light):
+    """Opple light device."""
+
+    def __init__(self, name, host):
+        """Initialize an Opple light."""
+        from pyoppleio.OppleLightDevice import OppleLightDevice
+        self._device = OppleLightDevice(host)
+
+        self._name = name
+        self._is_on = None
+        self._brightness = None
+        self._color_temp = None
+
+        _LOGGER.debug("init light %s %s", self._device.ip, self._device.mac)
+
+    @property
+    def available(self):
+        """Return True if light is available."""
+        return self._device.is_online
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._is_on
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def color_temp(self):
+        """Return the color temperature of this light."""
+        return kelvin_to_mired(self._color_temp)
+
+    @property
+    def min_mireds(self):
+        """Return minimum supported color temperature."""
+        return 175
+
+    @property
+    def max_mireds(self):
+        """Return maximum supported color temperature."""
+        return 333
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        _LOGGER.debug("Turn on light %s %s", self._device.ip, kwargs)
+        if not self.is_on:
+            self._is_on = self._device.power_on = True
```

Comment:
Let `update` change state holding instance attributes. It will be called directly after the method call.

Reply:
Change the line to:
```py
self._device.power_on = True
```
`is_on` will be updated in `update`.